pages/workflow/WorkflowHandler.inc.php

```diff
@@ -239,23 +245,21 @@ protected function getStageDecisionTypes(int $stageId): array
                 }
                 break;
             case WORKFLOW_STAGE_ID_EDITING:
-                /** @var ReviewRoundDAO $reviewRoundDao */
-                $reviewRoundDao = DAORegistry::getDAO('ReviewRoundDAO');
-                $hasReviewRound = $reviewRoundDao->submissionHasReviewRound($submission->getId(), WORKFLOW_STAGE_ID_EXTERNAL_REVIEW);
                 $decisionTypes = [
                     new SendToProduction(),
-                    $hasReviewRound
-                        ? new BackToReview()
-                        : new BackToSubmissionFromCopyediting()
                 ];
                 break;
             case WORKFLOW_STAGE_ID_PRODUCTION:
-                $decisionTypes = [
-                    new BackToCopyediting(),
-                ];
+                $decisionTypes = [];
                 break;
         }
 
+        $decisionTypes = array_merge(
+            $decisionTypes,
+            $decisionRepository->getApplicableRetractableDecisionTypes($stageId, $submission, $reviewRoundId),
+            $decisionRepository->getApplicableRemovableDecisionTypes($stageId, $submission, $reviewRoundId)
```

Comment:
I prefer to call `Repo::decision()` directly instead of assigning it to a variable. It makes it easier to search for code usage across the application, and it's not many characters.